ca/ca.go

```diff
@@ -262,38 +252,48 @@ func idForCert(cert *x509.Certificate) int64 {
 // from a single issuer (the first first in the issuers slice), and can sign OCSP
 // for any of the issuer certificates provided.
 func NewCertificateAuthorityImpl(
-	config ca_config.CAConfig,
 	sa certificateStorage,
 	pa core.PolicyAuthority,
-	clk clock.Clock,
-	stats prometheus.Registerer,
+	cfsslProfiles cfsslConfig.Config,
+	cfsslRSAProfile string,
+	cfsslECDSAProfile string,
 	cfsslIssuers []Issuer,
-	boulderIssuers []issuance.IssuerConfig,
+	boulderIssuers []*issuance.Issuer,
+	certExpiry time.Duration,
+	certBackdate time.Duration,
+	serialPrefix int,
+	maxNames int,
+	ocspLifetime time.Duration,
 	keyPolicy goodkey.KeyPolicy,
-	logger blog.Logger,
 	orphanQueue *goque.Queue,
+	logger blog.Logger,
+	stats prometheus.Registerer,
+	clk clock.Clock,
```

Comment:
The only part of this diff which isn't really necessary; I've rearranged the order of arguments here (and order of fields in the corresponding struct above) so that they're grouped by kind: dependency services, issuers, other profile-ish vars, infrastructure, test-only.